internal/querycoord/util.go

```diff
@@ -30,15 +31,47 @@ func getCompareMapFromSlice(sliceData []int64) map[int64]struct{} {
 	return compareMap
 }
 
-func getVecFieldIDs(schema *schemapb.CollectionSchema) []int64 {
-	var vecFieldIDs []int64
-	for _, field := range schema.Fields {
-		if field.DataType == schemapb.DataType_BinaryVector || field.DataType == schemapb.DataType_FloatVector {
-			vecFieldIDs = append(vecFieldIDs, field.FieldID)
+func estimateSegmentSize(segmentLoadInfo *querypb.SegmentLoadInfo) int64 {
+	segmentSize := int64(0)
+
+	vecFieldID2IndexInfo := make(map[int64]*querypb.VecFieldIndexInfo)
+	for _, fieldIndexInfo := range segmentLoadInfo.IndexInfos {
+		if fieldIndexInfo.EnableIndex {
+			fieldID := fieldIndexInfo.FieldID
+			vecFieldID2IndexInfo[fieldID] = fieldIndexInfo
+		}
+	}
+
+	for _, fieldBinlog := range segmentLoadInfo.BinlogPaths {
+		fieldID := fieldBinlog.FieldID
+		if FieldIndexInfo, ok := vecFieldID2IndexInfo[fieldID]; ok {
+			segmentSize += FieldIndexInfo.IndexSize
+		} else {
+			segmentSize += GetFieldSizeFromFieldBinlog(fieldBinlog)
 		}
 	}
 
-	return vecFieldIDs
+	// get size of state data
+	for _, fieldBinlog := range segmentLoadInfo.Statslogs {
+		segmentSize += GetFieldSizeFromFieldBinlog(fieldBinlog)
+	}
+
+	// get size of delete data
+	for _, fieldBinlog := range segmentLoadInfo.Deltalogs {
+		segmentSize += GetFieldSizeFromFieldBinlog(fieldBinlog)
+	}
+
+	return segmentSize
+}
+
+func GetFieldSizeFromFieldBinlog(fieldBinlog *datapb.FieldBinlog) int64 {
```

Comment:
maybe shouldn't be exported, `getFieldSizeFromFieldBinlog`

Reply:
fixed